provision/acc_provision/apic_provision.py

```diff
@@ -526,32 +545,52 @@ def associate_aep(self):
                             ]
                         }
                     },
-                    {
-                        "infraGeneric": {
-                            "attributes": {
-                                "name": "default",
-                            },
-                            "children": [
-                                {
-                                    "infraRsFuncToEpg": {
-                                        "attributes": {
-                                            "tDn": "uni/tn-%s/ap-kubernetes/epg-kube-nodes" % (tn_name,),
-                                            "encap": "vlan-%s" % (kubeapi_vlan,),
-                                        }
-                                    }
-                                }
-                            ]
-                        }
-                    }
                 ]
             }
         }
+        if self.with_kubeapi_vlan:
+            kubeapi_vlan = self.config["net_config"]["kubeapi_vlan"]
+            data["infraAttEntityP"]["children"].append(
+                {
+                    "infraGeneric": {
+                        "attributes": {
+                            "name": "default",
+                        },
+                        "children": [
+                            {
+                                "infraRsFuncToEpg": {
+                                    "attributes": {
+                                        "tDn": "uni/tn-%s/ap-kubernetes/epg-kube-nodes" % (tn_name,),
+                                        "encap": "vlan-%s" % (kubeapi_vlan,),
+                                    }
+                                }
+                            }
+                        ]
+                    }
+                })
 
         base = '/api/mo/uni/infra/attentp-%s' % aep_name
         rsvmm = base + '/rsdomP-[uni/vmmp-%s/dom-%s].json' % (vmm_type, vmm_name)
         rsphy = base + '/rsdomP-[uni/phys-%s].json' % phys_name
         rsfun = base + '/gen-default.json'
-        return path, data, rsvmm, rsphy, rsfun
+
+        if self.associate_aep_to_nested_inside_domain:
```

Comment:
why is this different for CF?

Reply:
CF VMs (node EPG) actually use the nested-inside domain rather than a physical domain, so we need to associate the AEP with the nested-inside domain.